### ISSUE TITLE
Available power not updating

### ISSUE THREAD

**Issue by frodef** [epaulsen/energytariff#17]
I'm observing currently that the "Available power" entity is not updating since the start of the hour. The value remains stuck at the value of target_energy.

I am currently running on solar power, and therfore my "Energy used this hour" entity remains at zero, and has not been updated since the start of the hour. I am guessing this is why "Available power" is not updating: It is currently only updated when "Energy used this hour" is updated?

If so, I would suggest adding an update every minute or so, regardless of whether "Energy used this hour" is updated.

**Comment by epaulsen** [comment]
That is correct. This integration listens for whenever your mains power sensor changes it's value, and then re-calculates all sensors that this integration exposes.  

Ideally this should be fixed, however this is a rather major undertaking, which I don't have time to do myself.  
As I don't have solar panels installed, verifying the fix would not be easy either.
Contributions in the form of a pullrequest that fixes this are more than welcome :)

**Comment by y0din** [comment]
I do not have solar power so I am unable to test this scenario, but could you not just add the option for one more listener and have them both update available power if set and also adding a config entry for this new listener defining it as solar/battery power?

If defined it could just make it update available power but not subtract the available power since it's not using import power?

Just an idea that hit me when I read this issue, which might not need to trigger a rewrite, just reusage mostly of existing code.
